### ISSUE TITLE
Topgrade breaks Chocolatey v1.0.0 setup for licensed customers

### ISSUE THREAD

**Issue by wadoli** [r-darwish/topgrade#875]
## What did you expect to happen?
With the recent release of Version 1.0.0 of [Chocolatey](https://chocolatey.org/), **(only) licensed customers** get their Chocolatey setup broken.
As noted in the [release notes for the _Chocolatey Licensed Extension_ v4.0.0](https://docs.chocolatey.org/en-us/licensed-extension/release-notes#march-21-2022),
> When using Chocolatey Licensed Extension, the order of upgrading packages is very important! You should perform choco upgrade chocolatey.extension first, and Chocolatey will then ensure all required dependencies are in place. If you first attempt to do choco upgrade chocolatey, followed by choco upgrade chocolatey.extension, there will be complications. If you find yourself in this situation, reach out to the Support Team for assistance.

Topgrade currently doesn't seem to follow that order

## What actually happened?
After topgrade has updated choco to version 1.0.0, licensed customers get an error message along the line of

```
Chocolatey v1.0.0 Professional
The registered delegate for type IEnumerable<ICommand> threw an exception. Der Typ "chocolatey.infrastructure.app.commands.ChocolateyVersionCommand" in der Assembly "choco, Version=1.0.0.0, Culture=neutral, PublicKeyToken=[redacted]" konnte nicht geladen werden.
```
## Additional details
- This only applies to Windows users
- Users can **fix** the resulting problem themselves. [Joel says it best](https://discord.com/channels/778552361454141460/897095556466372629/955806321641390141): "Simplest way is to go into your `C:\ProgramData\chocolatey\extensions` folder, rename the `chocolatey` folder in there to something else, then do `choco upgrade chocolatey.extension` to get the newer extension version. Once you've confirmed it works, you can delete the renamed folder from the extensions directory if you want."

**Comment by r-darwish** [comment]
What a terrible design 🤦 
Is there easy way to check if licensed extension is installed?

**Comment by TheCakeIsNaOH** [comment]
@r-darwish This command should work: `choco list chocolatey.extension --local-only --exact --limit-output`.
It will output `chocolatey.extension|<version>` if the licensed extension is installed.

Alternatively, if the `$env:ChocolateyInstall\lib\chocolatey.extension` folder exists, then the licensed extension should be installed.

**Comment by gep13** [comment]
It may be a good idea to perform a command like the following:

```
choco upgrade all --except="'chocolatey'"
```

That way users are opting into any changes to Chocolatey, since there can and will be breaking changes to the application in the future.

**Comment by wadoli** [comment]
This problem has also been addressed by the Chocolatey team in version 1.0.1 of the Chocolatey CLI, see https://docs.chocolatey.org/en-us/choco/release-notes#march-24-2022

**Comment by r-darwish** [comment]
@wadoli thanks for updating. No action will be done in  Topgrade, then
